oe_corporate_site_info.module

```diff
@@ -6,3 +6,188 @@
  */
 
 declare(strict_types = 1);
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\SortArray;
+use Drupal\Core\Form\FormStateInterface;
+
+function oe_corporate_site_info_form_system_site_information_settings_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $corporate_site_info = \Drupal::configFactory()->get('oe_corporate_site_info.settings');
+
+  $form['oe_site_information'] = [
+    '#type' => 'details',
+    '#title' => t('Corporate site information'),
+    '#open' => TRUE,
+  ];
+
+  $site_owner =  $corporate_site_info->get('site_owner') ?? NULL;
```

Comment:
We don't need to do any of this if later during the submit process we store the value from the form_state directly.